sympy/stats/joint_rv.py

```diff
@@ -0,0 +1,176 @@
+"""
+Joint Random Variables Module
+
+See Also
+========
+sympy.stats.rv
+sympy.stats.frv
+sympy.stats.crv
+sympy.stats.drv
+"""
+
+from __future__ import print_function, division
+
+__all__ = ['marginal_density']
+
+from sympy import Basic, Lambda, sympify, S
+from sympy.concrete.summations import Sum
+from sympy.integrals.integrals import Integral
+from sympy.stats.rv import (ProductPSpace, RandomSymbol,
+        NamedArgsMixin, ProductDomain)
+from sympy.core.containers import Tuple
+
+class JointPSpace(ProductPSpace):
```

Comment:
There is a logical problem here: `ProductPSpace` is implemented for the category of product spaces of independent random variables. If you inherit `ProductPSpace`, you are meant to deal with a subset of products of independent pspaces. Unfortunately the joint pspaces are a separate category. This current class inheritance structure is not good.

Furthermore, SymPy convention requires the subclass to have the same arguments in the constructor as the superclass.

Reply:
One solution could be to make `ProductPSpace` abstract, and move the implementation of the independent pspaces to another subclass, like `IndependentProductPSpace`.

Reply:
Sorry, I wasn't aware of the convention. I will implement the abstract class as suggested.
Will this need to undergo the deprecation process?

Reply:
no, because the new version of SymPy is still not out. Furthermore, if the API is hidden (cannot be imported directly), we can break it.

Reply:
That's why I suggest to keep the API hidden until the end of the summer of code. Unhiding only takes 5 minutes.

Reply:
`JointPSpace` could also be made seperate from `ProductPSpace` altogether, by inheriting directly from `PSpace`